### PR TITLE
More error reporting from failed goals

### PR DESCRIPTION
**Pull request by jessitron**
[atomist:generated]

### CODE REVIEW

**Comment by jessitron** [comment]
clay, there's another place that the CloudFoundryBlueGreen deployer throws an object instead of an Error, but I couldn't find it. the "Bad authentication" error was an object.

File: src/api-helper/log/LoggingProgressLog.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export class LoggingProgressLog implements ProgressLog {
     }
 
     public write(pWhat: string) {
-        let what = pWhat;
```
Comment (johnsonr) [inline src/api-helper/log/LoggingProgressLog.ts]: I generally use a convention of the parameter having the natural name and the implementation using `<naturalName>ToUse`

File: src/internal/delivery/goals/support/reportGoalError.ts [hunk 2]
```diff
@@ -1,80 +0,0 @@
...
-    logger.error("RunWithLog on goal %s with implementation name '%s' caught error: %s",
-        goal.name, implementationName, err.message);
-    logger.error(err.stack);
-    progressLog.write("ERROR: " + err.message + "\n");
-
-    const interpretation = logInterpreter(progressLog.log);
-    // The executor might have information about the failure; report it in the channels
-    if (interpretation) {
-        if (!interpretation.doNotReportToUser) {
-            await reportFailureInterpretationToLinkedChannels(implementationName, interpretation,
```
Comment (johnsonr) [inline src/internal/delivery/goals/support/reportGoalError.ts]: I wonder if we should make reporting to channels configurable. What about reporting to an error channel?

File: src/pack/pcf/CloudFoundryApi.ts [hunk 3]
```diff
@@ -99,10 +99,9 @@ export class CloudFoundryApi {
             return result;
         }
         if (failureCondition(result)) {
-            throw {
-                name: "RetryUntilConditionFailure",
-                message: JSON.stringify(result.data),
-            } as Error;
+            const error = new Error(JSON.stringify(result.data));
```
Comment (johnsonr) [inline src/pack/pcf/CloudFoundryApi.ts]: Should this be safe?

**Pull request merged**